compiler/src/dotty/tools/dotc/semanticdb/ExtractSemanticDB.scala

```diff
@@ -145,15 +148,43 @@ object ExtractSemanticDB:
     val path = semanticdbPath(source)
     Using.Manager { use =>
       val in = use(Files.newInputStream(path))
-      val sin = internal.SemanticdbInputStream.newInstance(in)
-      val docs = TextDocuments.parseFrom(sin)
+      // val sin = internal.SemanticdbInputStream.newInstance(in)
+      val textDocuments = Empty.parseFrom(in)
+      val docsBytes = textDocuments.getUnknownFields().getField(TextDocuments.DOCUMENTS_FIELD_NUMBER).getLengthDelimitedList()
+      val docFields = Empty.parseFrom(docsBytes.get(0)).getUnknownFields()
+      if (source.file.name == "ValPattern.scala")
+        println(docFields)
+      //docMap.put(7, )
+
+      // val docs = TextDocuments.parseFrom(sin)
+
+      val bos = use(new ByteArrayOutputStream())
+      val sbos = internal.SemanticdbOutputStream.newInstance(bos)
+      val doc = TextDocument(diagnostics = diagnostics)
+      doc.writeTo(sbos)
+      sbos.flush()
+      val diagnosticsOnly = Empty.parseFrom(bos.toByteArray()).getUnknownFields()
+
+      val merged = docFields.toBuilder().mergeFrom(diagnosticsOnly).build()
+      // println(merged)
+      val field = Field.newBuilder().addLengthDelimited(merged.toByteString()).build()
+
+      val fields = textDocuments.getUnknownFields().toBuilder().mergeField(TextDocuments.DOCUMENTS_FIELD_NUMBER, field).build()
+      // println(fields)
+      val updated = textDocuments.toBuilder().setUnknownFields(fields).build()
```

Comment:
you should start from a fresh `TextDocuments` instead of reusing the existing one - the current state will be carried over in the unknown field